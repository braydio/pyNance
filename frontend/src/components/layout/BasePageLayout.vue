<template>
  <div
    class="base-page-layout"
    :style="{ padding: paddingStyle, gap: spacingStyle }"
  >
    <slot name="header" />
    <slot />
  </div>
</template>

<script setup>
import { computed } from 'vue'
/**
 * BasePageLayout provides a reusable page wrapper with common
 * background, text color, spacing, and padding. Header content can
 * be injected via the `header` slot while default slot renders the
 * body. Padding and spacing can be customized with props.
 */
const props = defineProps({
  // Optional padding applied to the page container
  padding: { type: [String, Number], default: '1.5rem' },
  // Optional vertical spacing between header and body elements
  spacing: { type: [String, Number], default: '1.5rem' },
})

const paddingStyle = computed(() =>
  typeof props.padding === 'number' ? `${props.padding}rem` : props.padding
)
const spacingStyle = computed(() =>
  typeof props.spacing === 'number' ? `${props.spacing}rem` : props.spacing
)
</script>

<style scoped>
.base-page-layout {
  background-color: var(--page-bg);
  color: var(--theme-fg);
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}
</style>
