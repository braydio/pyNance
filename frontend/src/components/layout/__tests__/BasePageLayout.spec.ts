// @vitest-environment jsdom
import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import BasePageLayout from '../BasePageLayout.vue'

describe('BasePageLayout', () => {
  it('uses default padding', () => {
    const wrapper = mount(BasePageLayout, {
      slots: { default: '<div>Content</div>' },
    })
    expect(wrapper.classes()).toContain('p-6')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('accepts custom padding', () => {
    const wrapper = mount(BasePageLayout, {
      props: { padding: 'p-4' },
      slots: { default: '<div>Content</div>' },
    })
    expect(wrapper.classes()).toContain('p-4')
    expect(wrapper.classes()).not.toContain('p-6')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('applies gap class when provided', () => {
    const wrapper = mount(BasePageLayout, {
      props: { gap: 4 },
      slots: { default: '<div>A</div><div>B</div>' },
    })
    expect(wrapper.classes()).toContain('gap-4')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
