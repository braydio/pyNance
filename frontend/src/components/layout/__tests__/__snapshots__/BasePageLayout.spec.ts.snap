// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BasePageLayout > renders with default padding and gap 1`] = `"<div class="flex flex-col p-6 gap-6"></div>"`;
